app/async_database.py

```diff
@@ -0,0 +1,165 @@
+import aiosqlite
+import discord
+from database import UserProfile
+
+
+class AsyncDatabase:
+    """Class to store user profile and channel data asynchronously."""
+
+    db_path = "demo.db"
+
+    async def setup(self) -> None:
+        """Init for all asynchronous work for database."""
+        async with aiosqlite.connect(self.db_path) as db:
```

Comment:
I think that the DB connection should be shared. This means having an async context manager to make the database:

```py
@contextlib.asynccontextmanager
async def open_database(path: str) -> typing.Generator[AsyncDatabase]:
  async with aiosqlite.connect(path) as db:
    yield AsyncDatabase(db)
```

Then AsyncDatabase can just have its init take the database connection and share that